crates/wal-protocol/src/lib.rs

```diff
@@ -46,18 +45,21 @@ impl Envelope {
         Self { header, command }
     }
 
+    #[cfg(feature = "serde")]
     pub fn to_bytes(&self) -> Result<Bytes, StorageEncodeError> {
         let mut buf = BytesMut::default();
         StorageCodec::encode(self, &mut buf)?;
         Ok(buf.freeze())
     }
 
+    #[cfg(feature = "serde")]
```

Comment:
Maybe we can have a dedicated impl block that is feature gated together with the imports. That way the bits of the `serde` feature are kept closer together.